forestdb/iterator.go

```diff
@@ -87,5 +88,8 @@ func (i *Iterator) Close() error {
 	if i.curr != nil {
 		i.curr.Close()
 	}
+	if i.parent != nil {
+		i.parent.Close()
```

Comment:
I have some serious concerns here.  Closing an iterator should not close the Reader.  In Bleve we often create multiple iterators from the same reader.  The closing of one iterator should not close the reader, and if it did it likely breaks lots of existing functionality.